cmd/template/dbdriver/files/service/mysql.tmpl

```diff
@@ -56,4 +63,13 @@ func (s *service) Health() map[string]string {
 	return map[string]string{
 		"message": "It's healthy",
 	}
+}
+
+// Close closes the database connection.
+// It logs a message indicating the disconnection from the specific database.
+// If the connection is successfully closed, it returns nil.
+// If an error occurs while closing the connection, it returns the error.
+func (s *service) Close() error {
+	log.Printf("Disconnected from database: %s", dbname)
+	return s.db.Close()
```

Comment:
wouldn't we want to try to close before logging? What happens if `s.db.Close` returns an error? Will it still have disconnected?

Reply:
> wouldn't we want to try to close before logging? What happens if `s.db.Close` returns an error? Will it still have disconnected?

The `log.Printf("Disconnected from database: %s", dbname)` statement is useful when used with goroutines, such as for asynchronously starting and shutting down the server. However, it's recommended to handle the error returned by `s.db.Close` before logging the disconnection message. This way, you can ensure that the database connection is properly closed, even if an error occurs during the process. If `s.db.Close` returns an error, it doesn't necessarily mean that the connection has been successfully disconnected.

Reply:
@briancbarrow example how it work its like this

```sh
$ go run cmd/api/main.go
2024/04/04 20:52:02 [H0llyW00dzZ Firewall] [INFO] Connected to database: defaultdb
2024/04/04 20:52:04 [H0llyW00dzZ] [INFO] Starting server on :8080

 ┌───────────────────────────────────────────────────┐ 
 │                    H0llyW00dzZ                    │ 
 │                   Fiber v2.52.4                   │ 
 │               http://127.0.0.1:8080               │ 
 │       (bound on host 0.0.0.0 and port 8080)       │ 
 │                                                   │ 
 │ Handlers ............ 537 Processes ........... 1 │ 
 │ Prefork ....... Disabled  PID ............. 15836 │ 
 └───────────────────────────────────────────────────┘ 

2024/04/04 20:52:12 [H0llyW00dzZ] [INFO] Shutting down server... reason: interrupt
2024/04/04 20:52:12 [H0llyW00dzZ] [INFO] Disconnected from database: defaultdb
2024/04/04 20:52:12 [H0llyW00dzZ] [INFO] Database connection closed.
2024/04/04 20:52:17 [H0llyW00dzZ] [INFO] Server exiting
```

Reply:
That My Fiber application is fully managed by goroutines running in parallel.